### ISSUE TITLE
Dropout does not modify tensor in place when requested

### ISSUE THREAD

**Issue by kodo-pp** [tenstorrent/tt-metal#30284]
### Describe the bug

`ttnn.experimental.dropout` does not modify the input tensor even if it is provided via the `output_tensor` argument. This means that dropout cannot be performed in place at the moment.

### Steps to reproduce the issue

Run the code:

```python
import ttnn
import torch
dev = ttnn.open_device(...)

tensor = ttnn.from_torch(torch.tensor([[1, 2, 3], [4, 5, 6]]), device=dev, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT)
output = ttnn.experimental.dropout(tensor, probability=0.5, scale=2.0, seed=12345, output_tensor=tensor)
print(tensor)
print(output)
```

`print(tensor)` prints the initial tensor, but `print(output)` prints the modified tensor.

### Expected behavior

`print(tensor)` should print the same tensor as `print(output)`, because dropout should have used it as the output tensor.

### Please complete the following environment information

- OS version: Ubuntu 22.04
- Python version: 3.10.12
- tt-metal KMD version: 2.3.0
- ttnn version: 0.63.0 (installed from `pip` in a `venv`)

**Comment by kodo-pp** [comment]
Need to add the EPAM_DX label.

**Comment by fahmedchanTT** [comment]
@ayerofieiev-tt Since this is a training related op - can you please take on ?

**Comment by ayerofieiev-tt** [comment]
@fahmedchanTT , lets see what claude can do here.
